capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -44,8 +44,10 @@
     "SELECT COUNT(card.study_set_id), study_set.id, study_set.title, study_set.description, " +
     "study_set.subject, university.name, user_info.user_name FROM study_set JOIN university " +
     "ON university.id = study_set.university_id JOIN user_info ON study_set.owner_id = user_info.id " +
-    "JOIN card ON card.study_set_id = study_set.id WHERE card.front ILIKE ? OR card.back ILIKE ? OR " +
-    "study_set.title ILIKE ? OR study_set.description ILIKE ? GROUP BY study_set.id, university.id, user_info.id";
+    "JOIN card ON card.study_set_id = study_set.id WHERE study_set.subject ILIKE ? OR university.name ILIKE ? OR " +
+    "study_set.title ILIKE ? OR study_set.description ILIKE ? OR user_info.user_name ILIKE ? " +
+    "GROUP BY study_set.id, university.id, user_info.id";
+  private final long number_of_placeholders = search_sql_statement.chars().filter(ch -> ch == '?').count();
```

Comment:
This doesn't need to be a class level constant. You can just get it before you reference it in the function.